src/lib.rs

```diff
@@ -243,15 +244,17 @@ impl Layer {
     /// let layer: Layer;
     /// # layer = unimplemented!();
     ///
-    /// let layer: *mut CAMetalLayer = layer.into_raw().cast();
-    /// // SAFETY: The pointer is a valid `CAMetalLayer` with +1 retain count.
+    /// let layer: *mut CAMetalLayer = layer.into_raw().as_ptr().cast();
+    /// // SAFETY: The pointer is a valid `CAMetalLayer`, and because we consumed `Layer` with
+    /// // `into_raw`, the pointer has +1 retain count.
     /// let layer = unsafe { Retained::from_raw(layer).unwrap() };
     ///
     /// // Use the `CAMetalLayer` here.
     /// ```
     #[inline]
-    pub fn into_raw(self) -> *mut c_void {
-        Retained::into_raw(self.layer).cast()
+    pub fn into_raw(self) -> NonNull<c_void> {
+        // Unwrap is fine, `Retained` stores a non-null pointer
+        NonNull::new(Retained::into_raw(self.layer).cast()).unwrap()
```

Comment:
Guessing (given your PR description here about `Box`) that you might plan on changing the return type of `Retained` too?

Reply:
Yeah, I'm considering it